x/lockup/spec/05_keeper.md

```diff
@@ -0,0 +1,36 @@
+<!--
+order: 5
+-->
+
+# Keepers
+
+## Lock Keeper
+
+Lock Keeper provide utility functions to query user's locks.
+
+```go
+// Keeper is the interface for lockup module keeper
+type Keeper interface {
+	// Return full balance of the module
+	GetModuleBalance(sdk.Context) sdk.Coins
+	// Return locked balance of the module
+	GetModuleLockedAmount(sdk.Context) sdk.Coins
+
+	// Returns whole balance deposited by the user which is not withdrawn yet
+	GetAccountPoolBalance(sdk.Context, sdk.AccAddress) sdk.Coins
+	// Return a locked amount that can't be withdrawn
+	GetAccountLockedAmount(sdk.Context, sdk.AccAddress) sdk.Coins
+
+	// Returns the total number of tokens of an account whose unlock time is beyond timestamp
+	GetAccountLockedPastTime(sdk.AccAddress, timestamp time.Time) (sdk.Coins | []types.PeriodLock)
+	// Same as GetAccountLockedPastTime but denom specific
+	GetAccountLockedPastTimeDenom(sdk.AccAddress, denom string, timestamp time.Time) (sdk.Coins | []types.PeriodLock)
+	// Get iterator for all locks of a denom token that unlocks after timestamp
+	IteratorAccountsLockedPastTimeDenom(denom string, timestamp time.Time) db.Iterator
+	// Returns all the accounts that locked coins for longer than time.Duration.  Doesn't matter how long is left until unlock.  Only based on initial locktimes
+	IteratorLockPeriodsDenom(denom, time.Duration) []types.PeriodLock
+	// @sunny, single account can have multiple period locks, and it seems to be not valid query
```

Comment:
What do you mean?  Why is it not a valid query?

Reply:
I think instead of providing denom for this, providing LockID would make sense. (Just have changed to LockID)

Reply:
Ohh sorry, I accidentally thought this was a comment on the above function, not the below one.

Yeah, you're right, the below one doesn't make sense.  Should just have the ability to query by LockID.
```
// Returns the length of the initial lock time when the lock was created
GetLock(int64 lockid) Lock
```